src/parachain/loans.ts

```diff
@@ -375,16 +319,139 @@ export class DefaultLoansAPI implements LoansAPI {
+        const availableCapacityMonetary = newMonetaryAmount(availableCapacity.toString(), underlyingCurrency);
+        return [totalLiquidityMonetary, availableCapacityMonetary];
+    }
+
+    async _getLoanAssetLendAndBorrowYearlyRewardAmount(
+        underlyingCurrencyId: InterbtcPrimitivesCurrencyId
+    ): Promise<[Big, Big]> {
+        const [lendRewardSpeed, borrowRewardSpeed] = (
+            await Promise.all([
+                this.api.query.loans.rewardSupplySpeed(underlyingCurrencyId),
+                this.api.query.loans.rewardBorrowSpeed(underlyingCurrencyId),
+            ])
+        ).map((rewardSpeed) => decodeFixedPointType(rewardSpeed));
+
+        const blockTime = this.api.consts.timestamp.minimumPeriod.toNumber() * 2;
+        const blocksPerYear = (86400 * 365 * 1000) / blockTime;
+        // @note could be refactored to compute APR in lib if we can get underlyingCurrency/rewardCurrency exchange rate,
+        // but is it safe to assume that exchange rate for btc/underlyingCurrency will be
+        // always fed to the oracle and available?
+
+        // Return rate per 1 UNIT of underlying currency and compute APR
+        // on UI where all exchange rates are available.
+
+        const lendRewardAmount = lendRewardSpeed.mul(blocksPerYear);
+        const borrowRewardAmount = borrowRewardSpeed.mul(blocksPerYear);
+
+        return [lendRewardAmount, borrowRewardAmount];
+    }
+
+    _getLoanAssetLoanReward(amount: Big, currency: CurrencyExt): LoanReward | null {
+        if (amount.eq(0)) {
+            return null;
+        }
+        return {
+            currency,
+            amountPerUnitYearly: newMonetaryAmount(amount, currency),
+        };
+    }
+
+    async _getLoanAsset(
+        underlyingCurrencyId: InterbtcPrimitivesCurrencyId,
+        marketData: PalletLoansMarket
+    ): Promise<[CurrencyExt, LoanAsset]> {
+        const underlyingCurrency = await currencyIdToMonetaryCurrency(
+            this.assetRegistryAPI,
+            this,
+            underlyingCurrencyId
+        );
+
+        const [
+            lendApy,
+            borrowApy,
+            [totalLiquidity, availableCapacity],
+            [lendRewardAmountYearly, borrowRewardAmountYearly],
+        ] = await Promise.all([
+            this._getLoanAssetLendApy(underlyingCurrencyId),
+            this._getLoanAssetBorrowApy(underlyingCurrencyId),
+            this._getLoanAssetTotalLiquidityAndCapacity(underlyingCurrency, underlyingCurrencyId),
+            this._getLoanAssetLendAndBorrowYearlyRewardAmount(underlyingCurrencyId),
+        ]);
+
+        const liquidationThreshold = decodeFixedPointType(marketData.liquidationThreshold);
+        const collateralThreshold = decodeFixedPointType(marketData.collateralFactor);
+
+        const lendReward = this._getLoanAssetLoanReward(lendRewardAmountYearly, underlyingCurrency);
+        const borrowReward = this._getLoanAssetLoanReward(borrowRewardAmountYearly, underlyingCurrency);
```

Comment:
I think this was meant to be
```suggestion
        const borrowReward = this._getLoanAssetBorrowReward(borrowRewardAmountYearly, underlyingCurrency);
```

Reply:
Actually, this method is supposed to only construct `SubsidyReward` (formerly `LoanReward`) object. It was not very clear, so I renamed the method to `_constructSubsidyReward`